test/testvalueflow.cpp

```diff
@@ -3202,6 +3203,38 @@ class TestValueFlow : public TestFixture {
         ASSERT_EQUALS(true, values.front().isPossible() || values.back().isPossible());
         ASSERT_EQUALS(true, values.front().intvalue == 0 || values.back().intvalue == 0);
     }
+
+    void valueFlowTerminatingCond() {
+        const char* code;
+
+        code = "void f(int i, int j) {\n"
+               "    if (i > j) return;\n"
+               "    bool x = i > j;\n"
+               "    bool b = x;\n"
+               "}\n";
+        ASSERT_EQUALS(true, testValueOfX(code, 4U, 0));
+
+        code = "void f(int i, int j) {\n"
+               "    if (i > j) return;\n"
+               "    bool x = i < j;\n"
+               "    bool b = x;\n"
+               "}\n";
+        ASSERT_EQUALS(true, testValueOfX(code, 4U, 1));
+
+        code = "void f(int i) {\n"
+               "    if (i > 0) return;\n"
+               "    bool x = i > 0;\n"
+               "    bool b = x;\n"
+               "}\n";
+        ASSERT_EQUALS(true, testValueOfX(code, 4U, 0));
+
+        code = "void f(int i) {\n"
+               "    if (i > 0) return;\n"
+               "    bool x = i < 0;\n"
+               "    bool b = x;\n"
+               "}\n";
+        ASSERT_EQUALS(true, testValueOfX(code, 4U, 1));
+    }
```

Comment:
do you have false positive tests anywhere?

Modification of variable:

    if (i < j) return;
    i++;
    if (i > j) {} // <- not known

inner scope..

    if (a) {
        if (i < j) return;
    }
    if (i > j) {}  // <- not known

opposite order..

    if (i < j) {}  // <- not known
    if (i > j) return;